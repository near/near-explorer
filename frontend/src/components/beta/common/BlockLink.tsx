import * as React from "react";
import Link from "../../utils/Link";
import { styled } from "../../../libraries/styles";

export interface Props {
  blockHash: string;
  blockHeight: number;
}

const LinkWrapper = styled("a", {
  whiteSpace: "nowrap",
  cursor: "pointer",
});

const BlockLink: React.FC<Props> = React.memo(({ blockHash, blockHeight }) => (
  <Link href="/blocks/[hash]" as={`/blocks/${blockHash}`}>
    <LinkWrapper>{`#${blockHeight}`}</LinkWrapper>
  </Link>
));

export default BlockLink;
