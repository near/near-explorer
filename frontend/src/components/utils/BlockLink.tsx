import Link from "next/link";

export interface Props {
  blockHash: string;
  children?: React.ReactNode;
}

export default ({ blockHash, children }: Props) => (
  <Link href="/blocks/[hash]" as={`/blocks/${blockHash}`}>
    <a className="block-link">
      {children || `${blockHash.substring(0, 7)}...`}
    </a>
  </Link>
);
