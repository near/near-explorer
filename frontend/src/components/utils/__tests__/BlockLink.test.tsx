import renderer from "react-test-renderer";

import BlockLink from "../BlockLink";

describe("<BlockLink />", () => {
  it("renders", () => {
    expect(
      renderer.create(
        <BlockLink blockHash="H4CpspC1bqkykKG6dtCoGmLepvcyV2f2phfoqNAn5L2b" />
      )
    ).toMatchSnapshot();
  });
});
