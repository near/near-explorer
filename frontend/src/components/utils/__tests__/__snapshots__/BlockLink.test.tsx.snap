// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlockLink /> renders 1`] = `
<a
  className="block-link"
  href="/blocks/H4CpspC1bqkykKG6dtCoGmLepvcyV2f2phfoqNAn5L2b"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  H4CpspC...
</a>
`;
